File: Program Python/Motorcycle.py
from Vehicle import Vehicle

class Motorcycle(Vehicle):
    def __init__(self, plat_nomer, merk, tahun_produksi, warna, jenis_motor, kapasitas_tangki):
        super().__init__(plat_nomer, merk, tahun_produksi, warna)
        self.jenis_motor = jenis_motor
        self.kapasitas_tangki = kapasitas_tangki

    def get_jenis_motor(self):
        return self.jenis_motor

    def get_kapasitas_tangki(self):
        return self.kapasitas_tangki

    def get_description(self):
        return f"Jenis Motor: {self.jenis_motor}, Kapasitas Tangki: {self.kapasitas_tangki}"

File: Program Cpp/Motorcycle.cpp
#pragma once

#include <iostream>
#include <string>
#include <vector>
#include "Vehicle.cpp"

using namespace std;

class Motorcycle : public Vehicle
{
	// Class Motorcycle melakukan inheritance kepada kelas Vehicle

private:
	string jenis_motor;
	string kapasitas_tangki;
	// Variable yang dibutuhkan

public:
	Motorcycle(string plat_nomer, string merk, string tahun_produksi, string warna, string jenis_motor, string kapasitas_tangki) : Vehicle(plat_nomer, merk, tahun_produksi, warna){
		this->jenis_motor = jenis_motor;
		this->kapasitas_tangki = kapasitas_tangki;
	}
	// Contrutor

	string getJenisMotor(){
		return this->jenis_motor;
	}
	
	string getKapasitasTangki(){
		return this->kapasitas_tangki;
	}

	string getDescription(){
		return "Jenis Motor: " + this->jenis_motor + ", " + "Kapasitas Tangki: " + this->kapasitas_tangki;
	}
};